examples/static_set/mapping_table_example.cu

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuco/static_set.cuh>
+
+#include <cuda/std/array>
+#include <cuda/std/tuple>
+#include <thrust/device_vector.h>
+
+#include <iostream>
+
+/**
+ * @file mapping_table_example.cu
+ *
+ * @brief Demonstrates how to use hash set as a lookup table of the original data
+ *
+ * `cuco` hash tables such as `cuco::static_set` or `cuco::static_map` currently support only 4/8
```

Comment:
I'd say supporting only native typed keys is by-design rather than a missing feature.

Reply:
Updated the doc a bit based on offline discussions.